File: solutions/1302_delete-characters-to-make-fancy-string.java
// Problem: https://leetcode.com/problems/delete-characters-to-make-fancy-string
// Runtime: 28 ms

class Solution {
    public String makeFancyString(String s) {
        StringBuilder out = new StringBuilder("");
        char last = '_';
        int count = 0;
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (c == last) {
                count += 1;
            } else {
                count = 1;
            }
            if (count < 3) {
                out.append(c);
            }
            last = c;
        }
        return out.toString();
    }
}

File: solutions/1302_delete-characters-to-make-fancy-string.py
# Problem: https://leetcode.com/problems/delete-characters-to-make-fancy-string

class Solution:
    def makeFancyString(self, s: str) -> str:
        out = ""
        count = 1
        prev = ""

        for c in s:
            if c == prev:
                count += 1
            else:
                count = 1
            if count < 3:
                out += c
            prev = c
        
        return out